client/my-sites/sidebar/index.jsx

```diff
@@ -542,18 +550,59 @@ export class MySitesSidebar extends Component {
 	}
 
 	plan() {
+		// Hide "Plans" because the App/Play Stores reject apps that present non In-App Purchase flows, even in a WebView
+		if ( this.props.isWpMobile ) {
+			return null;
+		}
+
+		// Hide the plan name only for Jetpack sites that are not Atomic or VIP.
+		const displayPlanName = ! this.props.isJetpack || this.props.isAtomicSite || this.props.isVip;
+
+		/* eslint-disable wpcalypso/jsx-classname-namespace */
+		return (
+			<ExpandableSidebarMenu
+				onClick={ this.toggleSection( SIDEBAR_SECTION_PLAN ) }
+				expanded={ this.props.isPlanSectionOpen }
+				title={
+					<>
+						<span className="menu-link-text" data-e2e-sidebar="Plan">
+							{ this.props.translate( 'Plan', { context: 'noun' } ) }
+						</span>
+						{ displayPlanName && (
+							<span className="sidebar__menu-link-secondary-text">
+								{ this.props.site?.plan.product_name_short }
+							</span>
+						) }
+					</>
+				}
+				customIcon={ <Gridicon icon="star" className="sidebar__menu-icon" size={ 24 } /> }
+			>
+				{ this.plans() }
+				<SidebarItem
```

Comment:
@sirbrillig: did you intend to show this when the feature flag isn't present? I actually don't mind it redirection to `/me/purchases/` until this is live, but I'm curious to hear what @fditrapani thinks.

Reply:
Oops, that was not intentional. Thanks for noticing!